augur/filter/include_exclude_rules.py

```diff
@@ -898,8 +898,32 @@ def extract_variables(pandas_query: str):
     # covered by Python (unlikely, but I'm not sure). In those cases, don't
     # return anything.
     try:
-        return set(node.id
-                   for node in ast.walk(ast.parse(pandas_query))
-                   if isinstance(node, ast.Name))
+        modified_query, replacement_pair = _replace_backtick_quoting(pandas_query)
+        variables = set(node.id
+                        for node in ast.walk(ast.parse(modified_query))
+                        if isinstance(node, ast.Name))
+        for original_name, generated_name in replacement_pair:
+            variables.remove(generated_name)
+            variables.add(original_name)
```

Comment:
This assumes that every replaced backticked string will be extracted/identified as a variable. I don't think that's always the case, e.g. consider:

    column == "foo `bar` baz"

Since `set.remove()` throws a KeyError when the value isn't a member, this would cause errors (resulting in failure to optimize columns loaded).

So I'd think to do something more like:

```suggestion
for original_name, generated_name in replacement_pair:
    if generated_name in variables:
        variables.remove(generated_name)
        variables.add(original_name)
```

Reply:
Good catch. [Updated](https://github.com/nextstrain/augur/compare/7247039abda163fdeaad8975cc7cf4fbd881de69..8d85ffbb18e84d8ed518a0d11cc741a2139827f3) with the suggestion and added doctests for this scenario.